corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -946,7 +946,18 @@ def data_function(**view_kwargs):
         item_getter=None,
         event_handler=MigrationPaginationEventHandler(domain, stopper)
     )
-    return (row[row_key] for row in rows)
+    log.info("iteration state: %r", rows.state)
+    row = None
+    try:
+        for row in rows:
+            assert row['key'][0] == domain, row
```

Comment:
Is this the thing we talked about that will make the case diff process fail hard if it ever is about to process a case from the wrong domain?

Reply:
Correct